Description: Avoids to glob .dpkg-old files by specifically globbing what we
 want (this allows to filter out other undesired files).
Bugs: http://bugs.debian.org/579981
Origin: vendor
Forwarded: yes
Author: Didier Raboud <didier@raboud.com>
Last-Update: 2010-05-03

Index: usb-modeswitch.git/usb_modeswitch.tcl
===================================================================
--- usb-modeswitch.git.orig/usb_modeswitch.tcl	2010-05-03 09:04:22.000000000 +0200
+++ usb-modeswitch.git/usb_modeswitch.tcl	2010-05-03 10:57:21.000000000 +0200
@@ -164,7 +164,9 @@
 # which would point to a possible ambiguity. If so, check if
 # SCSI values are needed
 
-set configList [glob -nocomplain $dbdir/$usb(idVendor):$usb(idProduct)*]
+# The glob matches $idVendor:$idProduct without postfix and with :spec=value postfixes
+# This allows to filter out .dpkg-old files
+set configList [glob -nocomplain $dbdir/$usb(idVendor):$usb(idProduct){,\[:a-Z0-9=\]*}]
 if {[llength $configList] == 0} {
 	Log "Aargh! Config file missing for $usb(idVendor):$usb(idProduct)! Exiting"
 	SafeExit
@@ -279,7 +281,9 @@
 # end and provide a fallback
 
 set report {}
-set configList [glob -nocomplain $dbdir/$usb(idVendor):$usb(idProduct)*]
+# The glob matches $idVendor:$idProduct without postfix and with :spec=value postfixes
+# This allows to filter out .dpkg-old files
+set configList [glob -nocomplain $dbdir/$usb(idVendor):$usb(idProduct){,\[:a-Z0-9=\]*}]
 foreach configuration [lsort -decreasing $configList] {
 	Log "checking config: $configuration"
 	if [MatchDevice $configuration] {
